docs/database/mongodb-integration.md

```diff
@@ -1,137 +1,310 @@
+{
+    // Use client...
+}
+```
+
+The `IMongoClient` is used to interact with the MongoDB server resource. It can be used to create databases that aren't already known to the app host project. When you define a MongoDB database resource in your app host, you could instead require that the dependency injection container provides an `IMongoDatabase` instance. For more information on dependency injection, see [.NET dependency injection](/dotnet/core/extensions/dependency-injection).
+
+### Add keyed MongoDB client
+
+There might be situations where you want to register multiple `IMongoDatabase` instances with different connection names. To register keyed MongoDB clients, call the <xref:Microsoft.Extensions.Hosting.AspireMongoDBDriverExtensions.AddKeyedMongoDBClient*> method:
+
+```csharp
+builder.AddKeyedMongoDBClient(name: "mainDb");
+builder.AddKeyedMongoDBClient(name: "loggingDb");
+```
 
-var myService = builder.AddProject<Projects.MyService>()
-                       .WithReference(mongodb);
+> [!IMPORTANT]
+> When using keyed services, it's expected that your MongoDB resource configured two named databases, one for the `mainDb` and one for the `loggingDb`.
+
+Then you can retrieve the `IMongoDatabase` instances using dependency injection. For example, to retrieve the connection from an example service:
+
+```csharp
+public class ExampleService(
+    [FromKeyedServices("mainDb")] IMongoDatabase mainDatabase,
+    [FromKeyedServices("loggingDb")] IMongoDatabase loggingDatabase)
+{
+    // Use databases...
+}
 ```
 
-If using MongoDB Atlas, you don't need the call to `.AddDatabase("mongodb");` as the database creation will automatically be handled by Atlas.
+For more information on keyed services, see [.NET dependency injection: Keyed services](/dotnet/core/extensions/dependency-injection#keyed-services).
 
-## Configuration
+### Configuration
 
 The .NET Aspire MongoDB database integration provides multiple configuration approaches and options to meet the requirements and conventions of your project.
 
-### Use a connection string
+#### Use a connection string
 
 When using a connection string from the `ConnectionStrings` configuration section, you can provide the name of the connection string when calling `builder.AddMongoDBClient()`:
 
 ```csharp
-builder.AddMongoDBClient("MongoConnection");
+builder.AddMongoDBClient("mongo");
 ```
 
-And then the connection string will be retrieved from the `ConnectionStrings` configuration section:
-
-Consider the following MongoDB example JSON configuration:
+The connection string is retrieved from the `ConnectionStrings` configuration section. Consider the following MongoDB example JSON configuration:
 
 ```json
 {
   "ConnectionStrings": {
-    "MongoConnection": "mongodb://server:port/test",
+    "mongo": "mongodb://server:port/test",
   }
 }
 ```
 
-Consider the following MongoDB Atlas example JSON configuration:
+Alternatively, consider the following MongoDB Atlas example JSON configuration:
```

Comment:
IMO - this is a little confusing why we are pointing to MongoDB Atlas.

Reply:
One of the engineers/advocates from MongoDB actually added this a while back. I think they just wanted to drive home the fact that you can still point to Atlas.

Reply:
It kind of feels like an ad to me.